src/PJ_lccm.c

```diff
@@ -0,0 +1,149 @@
+
+#define PJ_LIB__
+#include <errno.h>
+#include "proj.h"
+#include "projects.h"
+#include "proj_math.h"
+
+PROJ_HEAD(lccm, "Lambert Conformal Conic (2SP Michigan)")
+    "\n\tConic, Sph&Ell\n\tlat_1= and lat_2= or lat_0, scale=";
+
+# define EPS10  1.e-10
+
+struct pj_opaque {
+    double phi1;
+    double phi2;
+    double n;
+    double rho0;
+    double c;
+    int    ellips;
+    double scale;
+};
+
+
+static XY e_forward (LP lp, PJ *P) {          /* Ellipsoidal, forward */
+    XY xy = {0.0,0.0};
+    struct pj_opaque *Q = P->opaque;
+    double rho;
+
+    if (fabs(fabs(lp.phi) - M_HALFPI) < EPS10) {
+        if ((lp.phi * Q->n) <= 0.) {
+            proj_errno_set(P, PJD_ERR_TOLERANCE_CONDITION);
+            return xy;
+        }
+        rho = 0.;
+    } else {
+        rho = Q->c * (Q->ellips ? Q->scale * pow(pj_tsfn(lp.phi, sin(lp.phi),
+            P->e), Q->n) : pow(tan(M_FORTPI + .5 * lp.phi), -Q->n));
+    }
+    lp.lam *= Q->n;
+    xy.x = P->k0 * (rho * sin( lp.lam) );
+    xy.y = P->k0 * (Q->rho0 - rho * cos(lp.lam) );
+    return xy;
+}
+
+
+static LP e_inverse (XY xy, PJ *P) {          /* Ellipsoidal, inverse */
+    LP lp = {0.0,0.0};
+    struct pj_opaque *Q = P->opaque;
+    double rho;
+
+    xy.x /= P->k0;
+    xy.y /= P->k0;
+
+    xy.y = Q->rho0 - xy.y;
+    rho = hypot(xy.x, xy.y);
+    if (rho != 0.0) {
+        if (Q->n < 0.) {
+            rho = -rho;
+            xy.x = -xy.x;
+            xy.y = -xy.y;
+        }
+        if (Q->ellips) {
+            lp.phi = pj_phi2(P->ctx, pow(rho / (Q->scale * Q->c), 1./Q->n), P->e);
+            if (lp.phi == HUGE_VAL) {
+                proj_errno_set(P, PJD_ERR_TOLERANCE_CONDITION);
+                return lp;
+            }
+
+        } else
+            lp.phi = 2. * atan(pow(Q->c / rho, 1./Q->n)) - M_HALFPI;
+        lp.lam = atan2(xy.x, xy.y) / Q->n;
+    } else {
+        lp.lam = 0.;
+        lp.phi = Q->n > 0. ? M_HALFPI : -M_HALFPI;
+    }
+    return lp;
+}
+
+
+PJ *PROJECTION(lccm) {
+    double cosphi, sinphi;
+    int secant;
+    struct pj_opaque *Q = pj_calloc (1, sizeof (struct pj_opaque));
+
+    if (0==Q)
+        return pj_default_destructor (P, ENOMEM);
+    P->opaque = Q;
+
+    if (pj_param(P->ctx, P->params, "tscale").i)
+      Q->scale = pj_param(P->ctx, P->params, "dscale").f;
+    else
+      Q->scale = 1.;
+
+    Q->phi1 = pj_param(P->ctx, P->params, "rlat_1").f;
+    if (pj_param(P->ctx, P->params, "tlat_2").i)
+        Q->phi2 = pj_param(P->ctx, P->params, "rlat_2").f;
+    else {
+        Q->phi2 = Q->phi1;
+        if (!pj_param(P->ctx, P->params, "tlat_0").i)
+            P->phi0 = Q->phi1;
+    }
+    if (fabs(Q->phi1 + Q->phi2) < EPS10)
+        return pj_default_destructor(P, PJD_ERR_CONIC_LAT_EQUAL);
+
+    Q->n = sinphi = sin(Q->phi1);
+    cosphi = cos(Q->phi1);
+    secant = fabs(Q->phi1 - Q->phi2) >= EPS10;
+    if( (Q->ellips = (P->es != 0.)) ) {
+        double ml1, m1;
+
+        P->e = sqrt(P->es);
```

Comment:
Hi Kristian! Thanks for your feedback. Actually, this code is mostly just a copy of PJ_lcc.c, so I tried not to touch it all, since, as you mention in the issue #940, the only difference between LCCM and LCC is that scaling factor which appears in three places in the code. I haven't even touched any stylistic issues which I found, trying to keep the diff between PJ_lcc and PJ_lccm to a bare minimum: just those three lines in question. 

Therefore to apply your proposals, it would be nice:

- either to reuse all that projection logic from normal LCC in LCCM (I'm not sure what is the best way to achieve this with existing code structure). We can probably create a common file and reuse it in both LCC and LCCM. Or we can just express LCC via LCCM with +scale=1.
- or to keep them in sync, i.e. to apply your proposed changes in both files. Keeping both of them would obviously increase the maintenance cost (people have to apply changes in two places and eventually someone will forget to do that).

Please let me know what is your preferred way. Perhaps I can do that refactoring work in a separate branch.

Reply:
Oh, I didn't realize that LCC_2SP Michigan was so close to LCC ! I'd definitely think that we should just merge them together. ie add the few lines needed to PJ_lcc.c and take into account the ellipsoidal scale in the e_forward and e_inverse. And update LCC documentation to properly document the LCC_2SP Michigan trick. What do you think @kbevers  ?

Reply:
+1 on merging the two. If I understand correctly the scale factor is main difference and that is definitely smarter to hand in one file. A clean up of the code is welcome :-)

Reply:
We probably don't need a dedicated lccm. EPSG distinguishes LCC_1SP and LCC_2SP, but we have a single PROJ method for both, so that would fit that model. The +scale != 1 will be the signature of LCC_2SP Michigan. But I don't have a strong opinion on this. I'll defer to @kbevers judgment

Reply:
Let's keep it in the same projection. Sounds like we would just introduce a bunch of duplicate code otherwise.

Reply:
Done